src/utils/sftp/handle-image-forms.ts

```diff
@@ -0,0 +1,178 @@
+    );
+
+    try {
+      // Safely extract file extension
+      const fileType = file.type.split("/");
+      const extension = fileType.length > 1 ? fileType[1] : "png";
+
+      //Create a new file with timestamp to invalidate nextjs cache
+      const imageUrl = await handleCreateSftpFile({
+        dir: "images",
+        file,
+        filename: `${name}-${file.lastModified}.${extension}`,
+        isPublic,
+        overwrite,
+      });
+
+      if (!imageUrl) {
+        toast.error("Kunde inte ladda upp bilden. Försök igen senare.");
+        return { success: false, error: "Failed to upload image" };
+      }
+
+      // If there was a previous image and it's different, delete the old one
+      if (
+        oldImageUrl &&
+        oldImageUrl !== imageUrl &&
+        oldImageUrl.startsWith(env.NEXT_PUBLIC_SFTP_BASE_URL)
+      ) {
+        try {
+          await handleDeleteSftpFile({ url: oldImageUrl });
+        } catch (deleteError) {
+          // Log but don't block the update if old image deletion fails
+          console.error("Failed to delete old image:", deleteError);
+        }
+      }
+
+      return { success: true, data: imageUrl };
+    } catch (error) {
+      if (error instanceof Error) {
+        toast.error(`Bilduppladdning misslyckades: ${error.message}`);
+        return { success: false, error: error.message };
+      } else {
+        toast.error("Något gick fel vid bilduppladdning. Försök igen senare.");
+        return { success: false, error: "Unknown error during upload" };
+      }
+    } finally {
+      toast.dismiss(loadingToastId);
+    }
+  },
+
+  /**
+   * Removes an existing image from the server
+   */
+  removeImage: async ({
+    imageUrl,
+  }: {
+    imageUrl: string;
+  }): Promise<ImageOperationResult<boolean>> => {
+    const loadingToastId = toast.loading(
+      "Tar bort bilden. Detta kan ta en stund!",
+    );
+
+    try {
+      await handleDeleteSftpFile({ url: imageUrl });
+      return { success: true, data: true };
+    } catch (error) {
+      if (error instanceof Error) {
+        toast.error(`Borttagning av bild misslyckades: ${error.message}`);
+        return { success: false, error: error.message };
+      } else {
+        toast.error(
+          "Något gick fel vid borttagning av bild. Försök igen senare.",
+        );
+        return { success: false, error: "Unknown error during removal" };
+      }
+    } finally {
+      toast.dismiss(loadingToastId);
+    }
+  },
+
+  /**
+   * Processes image state changes and returns the updated data
+   * Handles three cases:
+   * 1. New image being uploaded
+   * 2. Existing image being removed
+   * 3. No changes to image
+   */
+  processImageChanges: async ({
+    newImageFile,
+    currentImageUrl,
+    oldImageUrl,
+    entityName,
+  }: {
+    currentImageUrl?: string;
+    entityName: string;
+    newImageFile?: File | null;
+    oldImageUrl?: string;
+  }): Promise<ImageOperationResult<string | undefined>> => {
```

Comment:
currentImageUrl skickas alltid in. Denna kan därför göras icke optional och return type uppdateras
```suggestion
  processImageChanges: async ({
    newImageFile,
    currentImageUrl,
    oldImageUrl,
    entityName,
  }: {
    currentImageUrl: string;
    entityName: string;
    newImageFile?: File | null;
    oldImageUrl?: string;
  }): Promise<ImageOperationResult<string>> => {
```